pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -335,13 +335,12 @@ func (gc *GarbageCollector) attemptToDeleteWorker(ctx context.Context) bool {
 	}
 
 	err := gc.attemptToDeleteItem(ctx, n)
-	if err == enqueuedVirtualDeleteEventErr {
+	switch {
+	case err == enqueuedVirtualDeleteEventErr:
 		// a virtual event was produced and will be handled by processGraphChanges, no need to requeue this node
-		return true
-	} else if err == namespacedOwnerOfClusterScopedObjectErr {
+	case err == namespacedOwnerOfClusterScopedObjectErr:
```

Comment:
Why don't we forget in this case?  The sync loop here didn't have an error

Reply:
Got it, updated. PTAL thanks.